R/osdc-package.R

```diff
@@ -5,3 +5,8 @@
 #' @importFrom lifecycle deprecated
 ## usethis namespace: end
 NULL
+
+# Allows for using tidyverse functionality without triggering CRAN NOTES,
+# since CRAN doesn't know that packages like dplyr use NSE.
+# For more details, see https://rlang.r-lib.org/reference/dot-data.html#where-does-data-live-
+utils::globalVariables(".data")
```

Comment:
Yea, it's more of a very technical thing related to how CRAN runs checks on R code to make sure it is valid and correct. This is a bit of a "hack" to bypass CRAN's checks when programming with dplyr. For instance, a typical code like:

```r
dplyr::select(data_object, column_name1, column_name2)
```

Uses a thing called Non-Standard Evaluation (NSE; don't need to use quotes around the column names). In other programming languages, this wouldn't work, because they would complain that "no such object named column_name1 ... exists". That's because it exists in the data frame, not in the environment. NSE is wonderful when doing data analysis, but is very difficult to deal with programmatically. When we submit to CRAN, we can't use NSE within our functions (or at least, we have to use "tricks" to make it work). One way of using dplyr verbs within a CRAN package is instead doing:

```r
dplyr::select(data_object, .data$column_name1, .data$column_name2)
```

And combined with the code `utils::globalVariables(".data")`, which tells the R package (during CRAN checks), that we are declaring a placeholder object called `.data`, which allows us to continue writing dplyr code in a way that doesn't cause issues, but that is also easy for us too!

Hope that helps 😁

Reply:
Ah, that helps a lot! Thanks for explaining 🥳